src/Components/CreateImageWizard/CreateImageWizard.js

```diff
@@ -195,9 +196,163 @@ const onSave = (values) => {
+        const uploadRequest = imageRequest.upload_request;
+        let formState = {};
+
+        formState['image-name'] = composeRequest.image_name;
+
+        formState.release = composeRequest?.distribution;
+        // set defaults for target environment first
+        formState['target-environment'] = {
+            aws: false,
+            azure: false,
+            gcp: false,
+            'guest-image': false,
+        };
+        // then select the one from the request
+        // if the image type is to a cloud provider we use the upload_request.type
+        // or if the image is intended for download we use the image_type
+        let targetEnvironment;
+        if (uploadRequest.type === 'aws.s3') {
+            targetEnvironment = imageRequest.image_type;
+        } else {
+            targetEnvironment = uploadRequest.type;
+        }
+
+        formState['target-environment'][targetEnvironment] = true;
+
+        if (targetEnvironment === 'aws') {
+            formState['aws-account-id'] = uploadRequest?.options?.share_with_accounts[0];
+        } else if (targetEnvironment === 'azure') {
+            formState['azure-tenant-id'] = uploadRequest?.options?.tenant_id;
+            formState['azure-subscription-id'] = uploadRequest?.options?.subscription_id;
+            formState['azure-resource-group'] = uploadRequest?.options?.resource_group;
+        } else if (targetEnvironment === 'gcp') {
+            // parse google account info
+            // roughly in the format `accountType:accountEmail`
+            const accountInfo = uploadRequest?.options?.share_with_accounts[0];
+            const [ accountTypePrefix, account ] = accountInfo.split(':');
+
+            switch (accountTypePrefix) {
+                case 'user':
+                    formState['google-account-type'] = 'googleAccount';
+                    formState['google-email'] = account;
+                    break;
+                case 'serviceAccount':
+                    formState['google-account-type'] = 'serviceAccount';
+                    formState['google-email'] = account;
+                    break;
+                case 'group':
+                    formState['google-account-type'] = 'googleGroup';
+                    formState['google-email'] = account;
+                    break;
+                case 'domain':
+                    formState['google-account-type'] = 'domain';
+                    formState['google-domain'] = account;
+                    break;
+            }
+        }
+
+        // customizations
+        // packages
+        formState['selected-packages'] = composeRequest?.customizations?.packages?.map(packageName => {
+            const packageDescription = getPackageDescription(composeRequest?.distribution, imageRequest?.architecture, packageName);
+            return ({
+                name: packageName,
+                description: packageDescription
+            });
+        });
```

Comment:
I have done something wrong here but am unsure what exactly to do. This returns an object of:

```
{
  "name": "openssh",
  "description": Promise {
    <state>: "fullfilled",
    <value>: "An open source implementation of SSH protocol version 2"
  }
}
```

I am unsure exactly why. I would expect the async/await function to resolve to the value since the state is "fullfilled". But, it doesn't. I thought about awaiting the `getPackageDescription` call but then this would end up requiring await/async all up the function chain. 

Any thoughts on what I've done wrong here?

Reply:
An async function always returns a promise. Even if it's resolved on return.

Reply:
Ok so turns out `.map` doesn't play with async as I thought it would. In order to resolve the "Promises" `forEach` worked better.